src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -1002,6 +1002,7 @@ public void RemoveAnalyzerReference(string fullPath)
 
     private OneOrMany<string> GetMappedAnalyzerPaths(string fullPath)
     {
+        fullPath = Path.GetFullPath(fullPath);
```

Comment:
Is this still necessary, now that the full path is being passed in by the client?

Reply:
I believe this is needed.

Without this change, variable will be: 

`s_razorSourceGeneratorSdkDirectory`
```
"Sdks\\Microsoft.NET.Sdk.Razor\\source-generators\\"
```

`fullPath`
```
"C:\\Program Files\\dotnet\\sdk\\6.0.420\\Sdks\\Microsoft.NET.Sdk.Razor\\targets\\..\\\\source-generators\\Microsoft.NET.Sdk.Razor.SourceGenerators.dll"
```

`Path.DirectorySeparatorChar`
```
 92 '\\'

```
and the if condition below would be false and we won't go through getting analyzer references at all:
```
if (fullPath.LastIndexOf(s_razorSourceGeneratorSdkDirectory, StringComparison.OrdinalIgnoreCase) + s_razorSourceGeneratorSdkDirectory.Length - 1 ==
    fullPath.LastIndexOf(Path.DirectorySeparatorChar))
```

Reply:
Oh, that's a surprising value for fullPath. I thought this code also ran in VS, so I'm confused as to how it works there.